packages/design-system/src/components/contextMenu/menu.js

```diff
@@ -113,33 +128,59 @@ const Menu = ({
         return;
       }
 
-      const focusableChildren = getFocusableChildren(menuRef.current);
+      const focusableChildren = getFocusableChildren(
+        menuRef.current,
+        isSubMenu
+      );
 
-      const prevIndex = focusableChildren.findIndex(
+      let prevIndex = focusableChildren.findIndex(
         (element) => element.id === focusedId
       );
 
       if (prevIndex === -1 && focusableChildren.length) {
         setFocusedId(focusableChildren[0].id);
-        return;
+        prevIndex = 0;
       }
 
-      const isAscending = [KEYS.UP, KEYS.LEFT].includes(key);
-      let newIndex = prevIndex + (isAscending ? -1 : 1);
+      // If we're moving up-down.
+      if ([KEYS.UP, KEYS.DOWN].includes(key)) {
+        const isAscending = KEYS.UP === key;
+        let newIndex = prevIndex + (isAscending ? -1 : 1);
 
-      if (newIndex === -1) {
-        newIndex = focusableChildren.length - 1;
-      }
+        if (newIndex === -1) {
+          newIndex = focusableChildren.length - 1;
+        }
 
-      // Otherwise move to the next element or loop around the list.
-      const newSelectedElement =
-        focusableChildren[newIndex % focusableChildren.length];
+        // Otherwise move to the next element or loop around the list.
+        const newSelectedElement =
+          focusableChildren[newIndex % focusableChildren.length];
 
-      newSelectedElement?.focus();
-      setFocusedId(newSelectedElement?.id || -1);
-      return;
+        newSelectedElement?.focus();
+        setFocusedId(newSelectedElement?.id || -1);
+        return;
+      }
+      // Maybe move from submenu to parent menu.
+      if (
+        isSubMenu &&
+        ((!isRTL && KEYS.LEFT === key) || (isRTL && KEYS.RIGHT === key))
+      ) {
+        // Get the button with expanded popup.
+        const parentButton = parentMenuRef.current.querySelector(
+          'button[aria-expanded="true"][aria-haspopup]'
```

Comment:
Hm, to be nitpicky, we don't just want the element to have the aria-haspopup property, as it (despite its `has` prefix) isn't a boolean property but an enum, so it should have the right property as well. But on the other hand, only something with a popup would have the expanded property, so we could actually just remove the last condition?

```js
          'button[aria-expanded="true"]'
```